### PR TITLE
feat: product variations matrix

### PR DESCRIPTION
**Pull request by rostyk-kanafotskyy**
## Type

* ### Feature
  Implements a new feature

## Description
Now we create a product type with an individual product request

*Please ensure commit messages follow [Angular commit conventions](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)*

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -113,8 +113,13 @@ exports.sourceNodes = async (
         childrenArray = product.relationships.children.data.map((rel) => rel.id)
       }
     }
-
-    const nodeData = {
+  
+    if (product.meta.variation_matrix) {
+      const { variation_matrix } = product.meta;
```
Comment (samblacklock) [inline gatsby-node.js]: Hey @rostyk-kanafotskyy do you run prettier in your IDE? The config should disallow those semi-colons 😄 👀  https://github.com/moltin/gatsby-source-moltin/blob/master/.prettierrc

**Review by samblacklock — CHANGES_REQUESTED** [review]

File: gatsby-node.js (hunk 1 above)
Comment (rostyk-kanafotskyy) [reply gatsby-node.js]: Thanks @samblacklock 
Fixing that

Comment (rostyk-kanafotskyy) [reply gatsby-node.js]: @samblacklock  Updated gatsby-node.js
 with prettier command

**Review by shaunmaharaj — APPROVED** [review]

**Review by samblacklock — APPROVED** [review]
LGTM 👍

**Pull request merged**